pydantic_ai_slim/pydantic_ai/ag_ui.py

```diff
@@ -362,6 +375,11 @@ async def run_ag_ui(
         ) as run:
             async for event in _agent_stream(run):
                 yield encoder.encode(event)
+            if on_complete is not None:
+                if _utils.is_async_callable(on_complete):
+                    await on_complete(run)
+                else:
+                    await _utils.run_in_executor(on_complete, run)
```

Comment:
@DouweM not sure this is what you meant...

```py
if _utils.is_async_callable(on_complete):
    await on_complete(run)
 ```  
 
 >"None" is not awaitable
  "None" is incompatible with protocol "Awaitable[_T_co@Awaitable]"
    "__await__" is not present
    
  And 
  
  ```py
  else:
       await _utils.run_in_executor(on_complete, run)
  ```
  
Is saying it's unreachable.